yt_dlp/extractor/rokfin.py

```diff
@@ -160,6 +169,157 @@ def _get_comments(self, video_id):
+        self.write_debug(f'Updated tokens: {self._access_mgmt_tokens.keys()}')
+        if {'access_token', 'expires_in', 'refresh_expires_in', 'refresh_token', 'token_type', 'id_token', 'not-before-policy', 'session_state', 'scope'} - self._access_mgmt_tokens.keys():
+            username, password = self._get_login_info()
+            self._logout()
+            self._perform_login(username, password)
+        authorization_hdr_val = try_get(self._access_mgmt_tokens, lambda tokens: tokens['token_type'] + ' ' + tokens['access_token'])
+        if authorization_hdr_val:
+            headers['authorization'] = authorization_hdr_val
+        return self._download_json(
+            url_or_request, video_id, note='Downloading JSON metadata' + (' [logged in]' if 'authorization' in headers else ''),
+            errnote='Unable to download JSON metadata' + (' [logged in]' if 'authorization' in headers else ''),
+            headers=headers, query=query)
+
+    def _get_OAuth_tokens(self):
+        PARTIAL_USER_CONSENT_URL_STEP_4_5 = urllib.parse.urlparse('https://secure.rokfin.com/auth/realms/rokfin-web/protocol/openid-connect/auth?client_id=web&redirect_uri=https%3A%2F%2Frokfin.com%2Fsilent-check-sso.html&response_mode=fragment&response_type=code&scope=openid&prompt=none')
+
+        # ---------------------------- CODE FLOW AUTHORIZATION ----------------------------
+        # https://openid.net/specs/openid-connect-core-1_0.html#CodeFlowAuth (Sec. 3.1.1)
+
+        # Steps 4 & 5 authorize yt-dlp to act on user's behalf.
+
+        # random_str() came from https://www.rokfin.com/static/js/2.*.chunk.js
+        def random_str():
+            rnd_lst = [random.choice('0123456789abcdef') for _ in range(36)]
+            rnd_lst[14] = '4'
+            rnd_lst[19] = '0123456789abcdef'[3 & ord(rnd_lst[19]) | 8]
+            rnd_lst[8] = rnd_lst[13] = rnd_lst[18] = rnd_lst[23] = '-'
+            return ''.join(rnd_lst)
+
+        user_consent_url_step_4_5 = PARTIAL_USER_CONSENT_URL_STEP_4_5._replace(query=urllib.parse.urlencode((lambda d: d.update(state=random_str(), nonce=random_str()) or d)(dict(urllib.parse.parse_qsl(PARTIAL_USER_CONSENT_URL_STEP_4_5.query))))).geturl()
+
+        # By making this HTTP request, the user authorizes yt-dlp to act on behalf of the user:
+        urlh = (self._download_webpage_handle(
+            user_consent_url_step_4_5, None, note='granting user authorization', errnote='user authorization rejected by Rokfin', fatal=False, encoding='utf-8') or (None, None))[1]
+
+        def parse_authorization_code(urlh):
+            codes = urllib.parse.parse_qs(urllib.parse.urldefrag(urlh.geturl()).fragment).get('code')
+            return codes[0] if codes else None
+
+        authorization_code = parse_authorization_code(urlh)
+        if not authorization_code:
+            self.write_debug('authorization failed: missing authorization code')
+            return None
```

Comment:
shouldn't this be a warning/error?

Reply:
Yes, it should. But are `--abort-on-error`/`--no-abort-on-error` relevant here? I know that, technically, these two only apply to download errors. Even so, can they be used in the current situation?

Reply:
No. Generally, we raise error for user+pass login. But if you think it makes sense to attempt extraction even with failed login, you can just give a warning instead